include/osm_lua_processing.h

```diff
@@ -177,10 +177,11 @@ class OsmLuaProcessing : public PbfReaderOutput {
 	uint64_t osmID;							///< ID of OSM object (relations have decrementing way IDs)
 	int64_t originalOsmID;					///< Original OSM object ID
 	WayID newWayID = MAX_WAY_ID;			///< Decrementing new ID for relations
-	bool isWay, isRelation;					///< Way, node, relation?
+	bool isWay, isRelation, isClosed;		///< Way, node, relation?
 
 	int32_t lon1,latp1,lon2,latp2;			///< Start/end co-ordinates of OSM object
 	OSMStore::handle_t nodeVec;				///< node vector
+	NodeVec *nodeVecPtr;
```

Comment:
You can actually obtain the lasted stored nodeVec also from the osmStore, using the nodeVec handle. But I think that it not actually needed. 

TBH i do not completely follow the flow of handle these different points/linestring/polygons, but I guess it makes it confusing that they are called from the to-be-defined lua script. 

But if you can use the nodeVecPtr instead of the handle, i guess that is prefered. And possibly, you can remove the handle all together ?

Reply:
Yes, we could probably do a bit of tidying here. I'll merge this for now just to fix the issue, but then look at tidying it later.

Reply:
I was not really sure if the nodeVecPtr stayed valid when other functions were called. So that is why I was using the handle to the last stored linestring in osm_store to call certain function:
````
https://github.com/kleunen/tilemaker/blob/master/src/osm_lua_processing.cpp#L204
````

But if you are sure nodeVecPtr stays valid, you can use that one also :)